### PR TITLE
Proposal writeup and Function for numerical plots

### LINKED ISSUE #4: Outline for the project proposal in README.md file in the root directory
Ideas and points to discuss for the project proposal
- a summary paragraph that describes the project at a high level
- a bulleted list of the functions to be implemented and high level description
- a paragraph describing where your packages fit into the Python ecosystem

### LINKED ISSUE #9: Exploratory Plots for numerical features
Exploratory Plots for numerical features:
input: x, y, color/fill, title, size of text, color-scheme; 
output: altair/matplot objects (e.g. Scatter plots))

### PR DESCRIPTION
**Pull request by deepaksidhu**

### CODE REVIEW

File: simpler_eda/numerical_eda.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    colour_scheme = "yellowgreenblue",
+    plot_width = 400,
+    plot_height = 300.
+    x_transform=False,
+    y_transform=False,
+):
+    """
+    This function takes in a data frame object, two numeric columns,
+    and produces either a scatter or line plot to visualize the
+    relationship of the two features. User can optionally change
```
Comment (nichowu) [inline simpler_eda/numerical_eda.py]: between the two features

File: simpler_eda/numerical_eda.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    default arguements for plot-type, color/fill, title, size of text,
```
Comment (nichowu) [inline simpler_eda/numerical_eda.py]: color not color/fill

File: simpler_eda/numerical_eda.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    color-scheme, and toggle log transformation for the x and y axis.
```
Comment (nichowu) [inline simpler_eda/numerical_eda.py]: colour

File: simpler_eda/numerical_eda.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    data : pandas.core.frame.DataFrame
+       Input dataframe object.
+    
+    xval : str
+      Variable used to represent the x-axis.
+    
+    yval : str
+      Variable used to represent the y-axis.
+    
+    plot_type : str
```
Comment (nichowu) [inline simpler_eda/numerical_eda.py:34]: specify: Options are "scatter" and "line".

File: simpler_eda/numerical_eda.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      Variable used to represent the graphical relationship between xval and yval.
+    
+    color : str, optional
+      Variable used to set the color of the marks in the plot object.
```
Comment (nichowu) [inline simpler_eda/numerical_eda.py]: Variable used to group data points in different colours.

File: simpler_eda/numerical_eda.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    
+    tilte : str, optional
+      Variable use to set the title of the plot.
```
Comment (nichowu) [inline simpler_eda/numerical_eda.py]: used

File: simpler_eda/numerical_eda.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    
+    font_size  : str, optional
```
Comment (nichowu) [inline simpler_eda/numerical_eda.py]: int, optional
... used to set ...

File: simpler_eda/numerical_eda.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+    x_transform : bool, optional
+      Determines whether a log transformation occurs on the x-axis.
+    
+    y_transform : bool, optional
+      Determines whether a log transformation occurs on the y-axis.
+    
+    Returns
+    -------
+    altair.vegalite.v3.api.Chart
```
Comment (nichowu) [inline simpler_eda/numerical_eda.py]: change it to:
`altair`

File: simpler_eda/numerical_eda.py [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+      Scatter plot or Line plot of user-specified variables,
+      along with summary statisitics such as mean, median and variance.
```
Comment (nichowu) [inline simpler_eda/numerical_eda.py]: might want to take it out?

File: README.md [hunk 10]
```diff
@@ -1,18 +1,31 @@
...
 ```bash
 $ pip install -i https://test.pypi.org/simple/ simpler_eda
 ```
 
-## Features
+## Functions
 
-- TODO
+This package contains three functions, each that accepts a pandas `DataFrame` for EDA. No specific data set is required for the functions to work. Each functions will have it's own required and optional arguements to configure the properties of the plot. 
+
```
Comment (nichowu) [inline README.md:19]: update description according to docstring (after updated PR) pls

File: README.md [hunk 11]
```diff
@@ -1,18 +1,31 @@
...
+3. `categorical_eda_plot:` This function takes in a data frame object, a numerical and a categorical column and produces either a jitter, violin, barchar and boxplot (two categoricals) for comparison. User can optionally change default arguements for plot-type, color/fill, title, size of text and color-scheme.
+
+## How the simpler_eda package fits into the Python ecosystem
+
+The simpler_eda package improves upon existing functions in the Altair packages library. Altair already includes many useful functions to visualize relationship between numerical and categorical features with the use of grammar of graphics syntax. But often this is quite cumbersome and prone to errors, simpler_eda package provides convenience by allowing user to perform EDA with a single line of code. There are a number of packages that already provides similar functionality in the Python Ecosystem, but most of them are not easily customizable. Some examples of the packages are [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews).
```
Comment (ChuckHo777) [inline README.md]:
```suggestion
The simpler_eda package improves upon existing functions in the Altair packages library. Altair already includes many useful functions to visualize the relationship between numerical and categorical features with the use of grammar of graphics syntax. But often this is quite cumbersome and prone to errors, simpler_eda package provides convenience by allowing users to perform EDA with a single line of code. There are a number of packages that already provide similar functionality in the Python Ecosystem, such as [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews). However, most of them are not easily customizable. Our simpler_eda package allows flexibility from plot types to plot titles.
```

Comment (nichowu) [inline README.md]: The simpler_eda package improves upon existing functions in the Altair packages library. There are a number of packages that already provides similar functionality in the Python Ecosystem, but most of them are not easily customizable. Some examples of the packages are [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews).

**Review by ChuckHo777 — CHANGES_REQUESTED** [review]
Edit the part regarding our package position in the python ecosystem.

File: README.md [hunk 12]
```diff
@@ -15,17 +15,17 @@ $ pip install -i https://test.pypi.org/simple/ simpler_eda
 
 ## Functions
 
-This package contains three functions, each that accepts a pandas `DataFrame` for EDA. No specific data set is required for the functions to work. Each functions will have it's own required and optional arguements to configure the properties of the plot. 
+This package contains three functions, each that accepts a pandas `DataFrame` for EDA. The EDA functions can be used with a dataset with numerical and categorical features. Each functions will have it's own required and optional arguments to configure the properties of the plot. 
 
-1. `correlation_eda_plot:` This function takes in a data frame object, a list of numerical features to be correlated (instead of all features) and outputs a correlation plot object. User can optionally change default arguements for color/fill, title, size of text and color-scheme.
+1. `correlation_eda_plot:`  Plot a correlation map with the given dataframe object and a list of numerical features. Users are allow to set multiple arguments regarding the setting of the correlation plot from color schemes, plot width, height and plot title.
```
Comment (ChuckHo777) [inline README.md]:
```suggestion
1. `correlation_eda_plot:`  Plot a correlation map with the given dataframe object and a list of numerical features. Users are allowed to set multiple arguments regarding the setting of the correlation plot from color schemes, plot width, height, and plot title.
```

File: README.md [hunk 13]
```diff
@@ -1,18 +1,31 @@
...
 $ pip install -i https://test.pypi.org/simple/ simpler_eda
 ```
 
-## Features
+## Functions
 
-- TODO
+This package contains three functions, each that accepts a pandas `DataFrame` for EDA. The EDA functions can be used with a dataset with numerical and categorical features. Each functions will have it's own required and optional arguments to configure the properties of the plot. 
+
+1. `correlation_eda_plot:`  Plot a correlation map with the given dataframe object and a list of numerical features. Users are allow to set multiple arguments regarding the setting of the correlation plot from color schemes, plot width, height and plot title.
```
Comment (nichowu) [inline README.md]: Users are **_allowed_** to set multiple arguments regarding the setting of the correlation plot **_including_** color schemes, plot width, height and plot title.

File: README.md [hunk 14]
```diff
@@ -15,17 +15,17 @@ $ pip install -i https://test.pypi.org/simple/ simpler_eda
...
 
-2. `numerical_eda_plot:` This function takes in a data frame object, two numeric columns, and produces either a scatter or line plot to visualize the relationship of the two features. User can optionally change default arguements for plot-type, color/fill, title, size of text, color-scheme and apply logarithm scaling on the x and y axis.
+2. `numerical_eda_plot:` This function takes in a data frame object, two numeric columns, and produces either a scatter or line plot to visualize the relationship between the two features. User can optionally change default arguments for plot-type, color, title, size of text, color-scheme, and toggle log transformation for the x and y axis.
```
Comment (ChuckHo777) [inline README.md]:
```suggestion
2. `numerical_eda_plot:` This function takes in a data frame object, two numeric columns, and produces either a scatter or line plot to visualize the relationship between the two numerical features. Users can optionally change default arguments for plot-type, color, title, size of text, color-scheme, and toggle log transformation for the x and y axis.
```

File: README.md [hunk 15]
```diff
@@ -15,17 +15,17 @@ $ pip install -i https://test.pypi.org/simple/ simpler_eda
...
 
-3. `categorical_eda_plot:` This function takes in a data frame object, a numerical and a categorical column and produces either a jitter, violin, barchar and boxplot (two categoricals) for comparison. User can optionally change default arguements for plot-type, color/fill, title, size of text and color-scheme.
+3. `categorical_eda_plot:`  This function takes in a data frame object and one categorical feature, to produce a histogram plot that visualizes the distribution of the feature. User can also choose to plot density graph of the feature by specifing in plot_type. The function also offers customization on color, plot title, font size, color-scheme, plot size, opacity level and facet factor.
```
Comment (ChuckHo777) [inline README.md]:
```suggestion
3. `categorical_eda_plot:`  This function takes in a data frame object and one categorical feature, to produce a histogram plot that visualizes the distribution of the feature. Users can also choose to plot the density graph of the feature by specifying in plot_type. The function also offers customization on color, plot title, font size, color-scheme, plot size, opacity level, and facet factor.
```

**Review by nichowu — CHANGES_REQUESTED** [review]
change requested.

File: README.md [hunk 16]
```diff
@@ -15,17 +15,17 @@ $ pip install -i https://test.pypi.org/simple/ simpler_eda
...
+The simpler_eda package improves upon existing functions in the Altair packages library. Altair already includes many useful functions to visualize the relationship between numerical and categorical features with the use of grammar of graphics syntax. But often this is quite cumbersome and prone to errors, simpler_eda package provides convenience by allowing users to perform EDA with a single line of code. There are a number of packages that already provide similar functionality in the Python Ecosystem, such as [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews). However, most of them are not easily customizable. Our simpler_eda package allows flexibility from plot types to plot titles.
```
Comment (ChuckHo777) [inline README.md]:
```suggestion
The simpler_eda package improves upon existing functions in the Altair packages library. Altair already includes many useful functions to visualize the relationship between numerical and categorical features with the use of grammar of graphics syntax. But often this is quite cumbersome and prone to errors, simpler_eda package provides convenience by allowing users to perform EDA with a single line of code. There are a number of packages that already provide similar functionality in the Python Ecosystem, such as [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews). However, most of them are not easily customizable. Our simpler_eda package allows flexibility from plot types, color scheme, to plot titles.
```

File: simpler_eda/numerical_eda.py [hunk 17]
```diff
@@ -3,52 +3,51 @@
...
+    plot_height=300,
     x_transform=False,
     y_transform=False,
 ):
     """
     This function takes in a data frame object, two numeric columns,
     and produces either a scatter or line plot to visualize the
-    relationship of the two features. User can optionally change
-    default arguements for plot-type, color/fill, title, size of text,
+    relationship between the two features. User can optionally change
```
Comment (ChuckHo777) [inline simpler_eda/numerical_eda.py]: please remember to update according to the description on readme if you agreed with my changes.

File: simpler_eda/numerical_eda.py [hunk 18]
```diff
@@ -3,52 +3,51 @@
...
-
 def numerical_plot(
     data,
-    xval = None,
-    yval = None,
-    plot_type = "scatter",
-    color = None,
-    title = None ,
-    font_size = 10,
-    colour_scheme = "yellowgreenblue",
```
Comment (ChuckHo777) [inline simpler_eda/numerical_eda.py]: Having a second thought, this is coloring for the different categories in a scatter plot. A sequential color frame may be hard to differentiate. tableau20 might be a good alternative.

**Review by ChuckHo777 — CHANGES_REQUESTED** [review]
Spelling check on description and a suggestion on the default value for color_scheme.

File: simpler_eda/numerical_eda.py (hunk 18 above)
Comment (deepaksidhu) [reply simpler_eda/numerical_eda.py]: Changed to categorical scheme

File: simpler_eda/numerical_eda.py (hunk 17 above)
Comment (deepaksidhu) [reply simpler_eda/numerical_eda.py]: Yes agreed with changes, I have implemented it.

File: README.md (hunk 13 above)
Comment (deepaksidhu) [reply README.md]: Good observation!

File: README.md (hunk 11 above)
Comment (deepaksidhu) [reply README.md]: Used Chuck's paragraph for this description.

File: README.md (hunk 10 above)
Comment (deepaksidhu) [reply README.md:19]: updated

File: simpler_eda/numerical_eda.py (hunk 9 above)
Comment (deepaksidhu) [reply simpler_eda/numerical_eda.py]: Applied, thanks!

**Review by ChuckHo777 — APPROVED** [review]
Great!

**Review by nichowu — APPROVED** [review]

**Pull request merged**